api/src/test/java/com/epam/pipeline/controller/configuration/ConfigurationControllerTest.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2017-2020 EPAM Systems, Inc. (https://www.epam.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.pipeline.controller.configuration;
+
+import com.epam.pipeline.config.JsonMapper;
+import com.epam.pipeline.controller.vo.configuration.RunConfigurationVO;
+import com.epam.pipeline.entity.configuration.RunConfiguration;
+import com.epam.pipeline.manager.configuration.RunConfigurationApiService;
+import com.epam.pipeline.test.creator.configuration.ConfigurationCreatorUtils;
+import com.epam.pipeline.test.web.AbstractControllerTest;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MvcResult;
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+
+@WebMvcTest(controllers = ConfigurationController.class)
+public class ConfigurationControllerTest extends AbstractControllerTest {
+
+    private static final long ID = 1L;
+    private static final String CONFIGURATION_URL = SERVLET_PATH + "/configuration";
+    private static final String CONFIGURATION_BY_ID_URL = CONFIGURATION_URL + "/%d";
+    private static final String ALL_CONFIGURATIONS_URL = CONFIGURATION_URL + "/loadAll";
+    private JsonMapper mapper;
+    private final RunConfiguration runConfiguration = ConfigurationCreatorUtils.getRunConfiguration();
+    private final RunConfigurationVO runConfigurationVO = ConfigurationCreatorUtils.getRunConfigurationVO();
+
+    @Autowired
+    private RunConfigurationApiService mockRunConfigurationApiService;
+
+    @Before
+    public void setUp() {
+        mapper = getObjectMapper();
+        mapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
```

Comment:
Unfortunately this doesn't work as you would expect. Basically `getObjectMapper()` doesn't return a new instance on each call.

To do this properly we have to use two instances of object mappers. Default one which is used for serialization / deserialization in spring cannot be used as is in tests. It can be used for serialization but not for deserealization. Therefore let's use two object mappers in the abstract class. Something like the following can be of help.

```java
final ObjectMapper serializationObjectMapper = currentObjectMapper;
final ObjectMapper deserializationObjectMapper = JsonMapper.newInstance();
deserializationObjectMapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
...
final Result<T> actualResult = JsonMapper.parseData(actual, typeReference, deserializationObjectMapper);
```